src/common/HTTPRequest.ts

```diff
@@ -150,9 +150,11 @@ export class HTTPRequest {
   private _headers: Record<string, string> = {};
   private _frame: Frame;
   private _continueRequestOverrides: ContinueRequestOverrides;
-  private _responseForRequest: Partial<ResponseForRequest>;
-  private _abortErrorReason: Protocol.Network.ErrorReason;
-  private _interceptResolutionState: InterceptResolutionState;
+  private _responseForRequest: Partial<ResponseForRequest> = {};
```

Comment:
I feel like this field is similar to `_abortErrorReason` from the usage. So, to have the same typing for the related two methods; would it make sense to return `undefined` for the case `responseForRequest` is called before `respond` call (like `abortForReason`); wdyt?

Also setting the default value to `{}` can mean a behavior change:
Calling `responseForRequest` before `respond` was returning `undefined` now it returns empty object.

So I think it makes sense to type this as `Partial<ResponseForRequest> | undefined` and not set a default value.

Reply:
Good point. Actually, I double checked the API doc and it says we promise to return a null here instead of undefined. So this one will be a breaking fix too.